### ISSUE TITLE
Validation does not work with dynamic elements

### ISSUE THREAD

**Issue by max-frai** [nvms/vue-atlas#90]
I have `Select` and `Input` built with `v-for` and a button with click handler. Inside handler I call `validateFields` on form. It works. But when list of `Input` or `Select` changes by reactivity, validation does not rely on them. So the fields could be empty but validation will pass.

**Comment by max-frai** [comment]
<img width="363" alt="default" src="https://user-images.githubusercontent.com/24208746/53192864-1fac2080-3618-11e9-80cd-a4ee3b56a5a5.png">
Illustration. First input is red because it was created before validation run. Second is ignored, it was added when `validate=true` inside validator.

**Comment by max-frai** [comment]
<img width="898" alt="default" src="https://user-images.githubusercontent.com/24208746/53193030-a19c4980-3618-11e9-8cf8-4423d90094c1.png">
Try it with example in docs, just add required for dynamic elements

**Comment by nvms** [comment]
thanks for catching this. fixed in a1ed7cf

**Comment by nvms** [comment]
nope. I broke it. I'll push a new fix shortly.

**Comment by nvms** [comment]
okay, actually fixed in 4290814 - will update docs as well

**Comment by max-frai** [comment]
Could you please publish new package with minor version change, please? And I will test everything

**Comment by nvms** [comment]
will do

**Comment by nvms** [comment]
this is done and docs have been updated

**Comment by max-frai** [comment]
There are still some problems. Now dynamic fields are highlighted but in some cases validation passes even when there are red inputs. I'm writing simple example, will show you soon

**Comment by nvms** [comment]
it sounds like you might not be passing a unique `name` to each of your dynamic components. this is critical for validation to work. each field being validated needs a unique `name`. peek at the source for the example at https://vue-atlas.com/documentation/form

in a nutshell:
```html
<va-form-item
    v-for="(item, index) in demoForm.dynamicItems"
    :label="item.label"
    :key="index">
    <va-input
      v-model="item.value"
      style="margin-right: 5px;"
      :rules="[{type:'required'}]"
      :name="'dynamic' + index"></va-input>
    <va-button @click="deleteItem(index)"><va-icon type="times"></va-icon></va-button>
  </va-form-item>
```

**Comment by max-frai** [comment]
You are right! Just tested it, unique names for all dynamic items did the job. But I found another issue with dynamic items.
Validate on new items, remove some items from list by reactivity. Now all fields are not empty, but validation flag is undefined. I expanded validation result and found that already deleted items are still in list for validation. I have 4 items of both `path` and `category`, but there are items for 6:

<img width="255" alt="default" src="https://user-images.githubusercontent.com/24208746/53201956-f26a6d00-362d-11e9-9c0c-e7c806f011c2.png">
